src/plugins/data_source/server/plugin.ts

```diff
@@ -53,5 +71,32 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
     return {};
   }
 
-  public stop() {}
+  public stop() {
+    this.dataSourceService!.stop();
+  }
+
+  private createDataSourceRouteHandlerContext = (
+    dataSourceService: DataSourceServiceSetup,
+    logger: Logger
+  ): IContextProvider<RequestHandler<unknown, unknown, unknown>, 'dataSource'> => {
+    return (context, req) => {
+      return {
+        opensearch: {
+          getClient: (dataSourceId: string) => {
+            try {
+              return dataSourceService.getDataSourceClient(
+                dataSourceId,
+                context.core.savedObjects.client
+              );
+            } catch (error: any) {
+              logger.error(
+                `Fail to get data source client for dataSourceId: [${dataSourceId}]. Detail: ${error.messages}`
```

Comment:
this may cause 500 error, let's make a note here to handle that and returns 4xx as a follow up

Reply:
Sure, I'll track that